specification/draft-2/README.md

```diff
@@ -0,0 +1,687 @@
+TODO: Intro
+
+ - CWL specifies a way of describing runnable command line tools and workflows.
+ - CWL descriptions are serialized as either YAML or JSON.
+ - A CWL description contains a formal declaration of the inputs and outputs of described process (using Apache Avro schema).
+
+
+# Core specification
+
+CWL specification comes in two parts:
+
+ - Core spec allows description of basic tools and workflows
+ - Extensions add features that enable describing more complex tools and workflows
+
+If using an extension feature, it needs to be explicitly listed in the requirements section
+of the description document. Vendors might also add features incompatible with rest of CWL -
+in this case, they should be explicitly listed as requirements with full URI as value of `class` property.
+
+
+## Command line tools
+
+Basic tool example:
+
+```yaml
+class: CommandLineTool
+label: Grep
+description: Filter lines of text file by regular expression.
+inputs:
+  - id: pattern
+    type: string
+    binding:
+      position: 1
+  - id: in_file
+    type: File
+    streamable: true
+    binding:
+      position: 2
+outputs:
+  - id: out_file
+    type: File
+    streamable: true
+    binding:
+      glob: output.txt
+baseCmd: grep
+stdout: output.txt
+successCodes: [0, 1]
+```
+
+Above is defined a command line tool labeled "grep" with two required inputs
+(a file and a string) and one file output.
+
+Input objects contain a `binding` section which describes how the arguments
+are passed onto the command line. In this case, `pattern` and `in_file` are
+positional arguments and `pattern` is passed first, followed by `in_file`.
+Identifiers for inputs and outputs must be unique across the document.
+
+The `binding` section of the output object contains a single property `glob`
+with a glob pattern describing how to recognize file generated by the tool.
+
+Additionally, `baseCmd` and `stdout` properties describe the executable name
```

Comment:
Yes, although in your example you could use `glob: "myfile"` in `outputBinding` to indicate that it it should be captured as an output file.  But it is preferrable to capture it explicitly:

```yaml
stdout: myfile
baseCmd: ["ls"]
```

The reasoning for being explicit (in addition to the desire to avoid dependence on the shell) is that it gives the workflow platform more options for managing `stdin`  and `stdout`, for example streaming to/from a remote storage service instead of staging it on the local disk.